### ISSUE TITLE
imporprivate key results in wrong balance

### ISSUE THREAD

**Issue by PSLLSP** [peercoin/peercoin#555]
Peercoin version v0.9.0ppc Strider (release buildersion: Peercoin version v0.9.0ppc Strider (release build)

I downloaded wallet and I synchronized it. Balance was 0.0.

I imported 1 private key with command `importprivkey  privatekeyissecret`. I use GUI, console in debug window.

Once the import was finished, GUI wallet reported that I have about 93 PPC in the wallet. That was good news because I expected to see only 57 PPC (and this is visible on public blockchain explorer). I verified with command `getbalance`, it reported 93 PPC.

I verified with command `listaddressgroupings`, it reported 57 PPC.

I exported transaction from GUI to text file and I used AWK to sum about 30 transactions transactions saved in export file.

```
awk '
BEGIN{FS=","; S=0;}
{B=$6; gsub(/"/,"",B); print B; S+=B;}
END{printf "SUM: %f\n", S;}' ppc-export.txt
```
This script reported 57 PPC.

I executed `rescanblockchain` command and after this `getbalance` reported 57 PPC and I see 57 PPC in GUI wallet. Finally, my coins are in order...

SUMMARY:
`getbalance` and GUI reports wrong balance after private key import. Transaction history in my wallet was correct. `listaddressgroupings` reported correct value. This issue was fixed with simple `rescanblockchain` without wallet restart.

**Comment by PSLLSP** [comment]
It is probably still the issue #529 that was reported several months ago and that will be fixed in wallet 0.9.1...

**Comment by PSLLSP** [comment]
I have one more note, it looks like new issue. I miss one transaction "Mint by stake" from Jun 2020 in the output from command `listransactions 9999 0`. The last transaction is missing. I see the transaction in the GUI, in the list of exported transactions.

I found this problem when I simply checked how many transaction  are in the json file and I noticed that one transaction is missing `jq 'length' listtransactions.json`

When I execute just `listtransactions`, it reports the last 10 transactions and "the missing" transaction is in the list...
